libr/asm/arch/wasm/wasm.c

```diff
@@ -184,11 +184,12 @@ static WasmOpDef opcodes[256] = {
 	[WASM_OP_F64REINTERPRETI64] = { "f64/reinterpret/i64", 1, 1 }
 };
 
+// assembles the given line of wasm assembly.
 R_IPI int wasm_asm(const char *str, unsigned char *buf, int buf_len) {
 	// TODO: add immediates assembly
 	int i = 0, len = -1;
-	char tmp[R_ASM_BUFSIZE];
-	while (str[i] != ' ' && i < buf_len) {
+	char tmp[256];
+	while (str[i] && str[i] != ' ' && i < 256) {
```

Comment:
use sizeof(tmp) instead of hardcoded 256

Reply:
oh, right